### ISSUE TITLE
[no-floating-promises] Change ignoreVoid to false in default ruleset?

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#2569]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking",
  ]
}
```

```TS
async function someAsyncFunction() { /* ... */ }

export function main() {
    void someAsyncFunction();
}
```

**Expected**

An error in the default configuration that `someAsyncFunction()`'s created promise is not handled.

**Actual**

`ignoreVoid` is turned on by default for `no-floating-promises`.

**Additional Info**

ESLint already has a `// eslint-disable-next-line` standard for ignoring rule violations. It seems odd to add a specific reimplementation of it just for this one rule _and_ recommend it as part of a prefab configuration.

Also, if you'll permit a self ad hominem _(is that the right fallacy descriptor...?)_, as the original implementer of the rule in TSLint I was very surprised to see users mention it as a need (https://github.com/palantir/tslint/issues/3020 -> https://github.com/palantir/tslint/issues/4653), as it was never an intention in the rule's original implementation. 

To be clear: although I am against the `ignoreVoid` operator as a whole (😉), the intent of _this_ issue is _only_ for changing its default to `false`.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.1.0` |
| `@typescript-eslint/parser`        | `4.1.0` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.7.0` |
| `node`                             | `14.7.0` |

**Comment by bradzacher** [comment]
disable comments can be cumbersome to use depending on your lint setup.

Most users don't like the wide and all suppressing `// eslint-disable-next-line`, so they instead list individual rules to narrow the scope ([there are lint rules against doing unlimited disables as well](https://mysticatea.github.io/eslint-plugin-eslint-comments/rules/no-unlimited-disable.html)).

This leads to the very long and cumbersome
```ts
// eslint-disable-next-line @typescript-eslint/no-floating-promises
someAsyncFunction();
```

On top of that many codebases don't like dangling disables in the codebase as it is never clear if the disable is there because it was a bug, or if it's entirely intentional, etc. (Again [there are lint rules against using disables without a description](https://mysticatea.github.io/eslint-plugin-eslint-comments/rules/require-description.html))
Which makes this more cumbersome in the form of:
```ts
// eslint-disable-next-line @typescript-eslint/no-floating-promises -- intentionally not awaited
someAsyncFunction();

// or pre-v7

// intentionally not awaited
// eslint-disable-next-line @typescript-eslint/no-floating-promises
someAsyncFunction();
```

OTOH, the `void` operator is rarely actually used in a codebase. Most people have `no-void` turned on. With a slight reconfiguration to turn on [`allowAsStatement`](https://eslint.org/docs/rules/no-void#allowasstatement), it means that there will only ever be a `void` operator as a statement.

Which means that this code is now unique in a codebase, and (IMO) very clear and very explicit:
```ts
void someAsyncFunction();
```

The feedback that I've seen is that it's not really a rare edge case to do this - it's just "uncommon", esp when writing entrypoints to nodejs code.
So relying upon comments is just a pain, and the `void` operator is a convenient way to explicitly do do it.

**Comment by JoshuaKGoldberg** [comment]
Hmm, everything in that response makes sense, but I can't shake the feeling that this is an odd one-of... I will rescind. Thanks for the detailed write-up!
